# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/html/HTMLCanvasElement.h
--- a/dom/html/HTMLCanvasElement.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/html/HTMLCanvasElement.h	Wed Jan 08 22:19:16 2020 +0000
@@ -339,8 +339,6 @@
 
   layers::OOPCanvasRenderer* GetOOPCanvasRenderer();
 
-  PWebGLChild* GetWebGLChild();
-
  protected:
   virtual ~HTMLCanvasElement();
 
@@ -371,8 +369,10 @@
                                           const nsAttrValueOrString& aValue,
                                           bool aNotify) override;
 
-  ClientWebGLContext* GetClientWebGLContext();
+ public:
+  ClientWebGLContext* GetWebGLContext();
 
+ protected:
   bool mResetLayer;
   bool mMaybeModified;  // we fetched the context, so we may have written to the
                         // canvas